### ISSUE TITLE
CI Server Builds

### ISSUE THREAD

**Issue by ylathouris** [ylathouris/glossy#1]
Add support for running an automated test suite when code changes are pushed. 

This should definitely happen for the  **master** branch and the **master** branch should always be in a working and deployable state. I think it makes sense to trigger the tests for other branches too. This should help us determine whether or not the changes are safe to merge into **master**. This isn't fool-proof though, because it won't tell us whether or not our changes work with changes in other branches but, at least not until those other changes are in the **master** branch.      

**Acceptance Criteria:**

- Automated tests run when code changes are pushed to remote branch
